userguide/tutorials/development.adoc

```diff
@@ -507,24 +375,129 @@ The steps above configure the application so that the Kill Bill logs are display
+
+.. Press Ctrl + Shift + T and locate `DefaultKillbillConfigSource`. 
+
+..  If this does not show the source code do the following:
+
+... Click *Attach Source*. 
+
+... Click on *External Location* and Navigate to your Maven local repository (`~/.m2/repository`). Select `/org/kill-bill/billing/killbill-platform-base/0.40.4/killbill-platform-base-0.40.4-sources.jar`
+
+.. Set up a breakpoint in `DefaultKillbillConfigSource.java#L118` 
+
+.. Click `Run > Debug Configurations`
+
+.. Double click  `New Remote Java Application`
+
+.. Enter the name that you would like to give to this debug configuration in the *Name* field
+
+.. Click *Apply*
+
+.. Click *Close*
+
+. Start the application as explained in *Running the Application* section above
+
+. Click `Run > Debug Configurations` and double click the  the Debug configuration that you created above
+
+. This runs the application in debug mode. so, you can set additional breakpoints and debug as required.
+
+=== Some common errors and their solutions
+
+==== Build Failure
+
+Sometimes, after building the application, the build may fail with the following errors:
+
+ [INFO] killbill-payment ................................... SUCCESS [01:42 min]
+ [INFO] killbill-beatrix ................................... SUCCESS [02:04 min]
+ [INFO] killbill-jaxrs ..................................... SUCCESS [01:45 min]
+ [INFO] killbill-profiles .................................. SUCCESS [  9.896 s]
+ [INFO] killbill-profiles-killbill ......................... FAILURE [02:48 min]
+ [INFO] killbill-profiles-killpay .......................... SKIPPED
+ [INFO] ------------------------------------------------------------------------
+ [INFO] BUILD FAILURE
+ [INFO] ------------------------------------------------------------------------
+ [INFO] Total time:  26:55 min
+ [INFO] Finished at: 2020-11-23T10:46:24+05:30
+ [INFO] ------------------------------------------------------------------------
+ [ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.13:check (default) on project killbill-profiles-killbill: Too many files with unapproved license: 1 See RAT report in: <PROJECT_ROOT>\profiles\killbill\target\rat.txt -> [Help 1]
+ [ERROR]
+ [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
+ [ERROR] Re-run Maven using the -X switch to enable full debug logging.
+ [ERROR]
+ [ERROR] For more information about the errors and possible solutions, please read the following articles:
+ [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
+ [ERROR]
+ [ERROR] After correcting the problems, you can resume the build with the command
+ [ERROR]   mvn <args> -rf :killbill-profiles-killbill
+ 
+This error is most probably because the `logback.xml` file is missing license information. This might have got deleted while customizing the log file path. So ensure that the `logback.xml` file has the following license information at the top:
```

Comment:
You can actually specify `-Dcheck.skip-rat=true` on the command line. Might be easier to do so than adding the license back.

Reply:
Changed